j2cl-tasks/src/main/java/com/vertispan/j2cl/build/provided/ClosureTask.java

```diff
@@ -163,7 +163,9 @@ public Task resolve(Project project, Config config) {
                         // Bytecode sources will include original input sources
                         // as well as generated input when the jar was built
                         input(p, OutputTypes.BYTECODE)
-                ));
+                ))
+                // Removing any duplicate dependencies that would otherwise trigger unjustified duplicate paths detection in Closure
+                .distinct();
```

Comment:
I don't really like doing this, for two chief reasons:
 * Since it seems like there could be more than one that might be the "right" answer expected by different libraries on a classpath. The BUNDLE_JAR tasks have a certain order (both of files in a maven module, and maven modules in the tree) already so that a distinct() will probably do the _opposite_ of what you intended, but module and file order here is not specified at all, is likely to be maven- and/or filesystem-specific.
 * It is currently convenient that modules are mostly 1:1 with paths, but that is mostly happenstance, and not something to rely on. 

Questions to consider:
 * Closure module de-duping has nothing to do with the module name - path similarity doesn't bother closure, only module names, so in the long run this isn't even fixing the right thing...?
 * This isn't necessarily ordered though right? How are you certain that this removes the "right" files?
 * Also, I'm pretty sure you also need this in the BUNDLE_JAR wiring so that j2cl:watch still works and doesnt break at runtime (which is how it responds to duplicate modules).


I think it probably should be redone in your own fork or your own task (note that taskfactory instances are picked up via service loader, so this is well supported...) as "this is a specific fix needed for my own build", or redone in this patch as "this dedup's `goog.provides` (by name), `goog.modules` (by name), and es6 modules (by path), not dedup all files by path".